pkg/controller/deployment/sync.go

```diff
@@ -189,9 +189,11 @@ func (dc *DeploymentController) getNewReplicaSet(ctx context.Context, d *apps.De
 	newRSTemplate := *d.Spec.Template.DeepCopy()
 	podTemplateSpecHash := controller.ComputeHash(&newRSTemplate, d.Status.CollisionCount)
 	newRSTemplate.Labels = labelsutil.CloneAndAddLabel(d.Spec.Template.Labels, apps.DefaultDeploymentUniqueLabelKey, podTemplateSpecHash)
+	newRSTemplate.Labels[apps.DeploymentUniqueLabelKey] = podTemplateSpecHash
 	// Add podTemplateHash label to selector.
 	newRSSelector := labelsutil.CloneSelectorAndAddLabel(d.Spec.Selector, apps.DefaultDeploymentUniqueLabelKey, podTemplateSpecHash)
```

Comment:
actually, can we remove the old key from the selector?
This means that the RS selector for new RS is independent of the legacy label, but users can still use the legacy label for a bit, if they are not paying attention to the selector.

We only need to make sure that we wouldn't recreate a replicaset just because it doesn't have the new label.
I'm not fully familiar with deployment controller to see if these changes are enough. Do we need any changes in FindNewReplicaSet?

We did the same for the job controller https://github.com/kubernetes/kubernetes/pull/114930

Reply:
Updted. I think in `FindNewReplicaSet` we compare two rs without hash label, so it should be ok. But I'm not sure as I'm not familiar with deployment controller.

Reply:
yes, we are not using the hashes for looking up/identifying the RSes, so it should be fine to use only the new label in the selector. The only potential problem I am seeing is, if somebody orphaned their pods and wanted to move them between replicasets while bringing this change in.